### PR TITLE
Deprecate more usage of QuantityType

### PR DESCRIPTION
**Pull request by tmilnthorp**
I noticed we had only partially deprecated QuantityType. I think this hits all of them.

### CODE REVIEW

**Comment by tmilnthorp** [comment]
Continued from #864

File: CodeGen/Generators/UnitsNetWrcGen/QuantityGenerator.cs [hunk 1]
```diff
@@ -585,7 +587,7 @@ public bool Equals({_quantity.Name} other, double tolerance, ComparisonType comp
         /// <returns>A hash code for the current {_quantity.Name}.</returns>
         public override int GetHashCode()
         {{
-            return new {{ QuantityType, Value, Unit }}.GetHashCode();
```
Comment (tmilnthorp) [inline CodeGen/Generators/UnitsNetWrcGen/QuantityGenerator.cs:588]: Matching implementation in WRC

**Comment by tmilnthorp** [comment]
CodeCov is really hating me today.

Love the error for a deleted/unused line of code: "Your changes would remove code coverage for this line" 😄

File: UnitsNet/GeneratedCode/Quantity.g.cs [hunk 2]
```diff
@@ -152,124 +152,6 @@ public static partial class Quantity
...
-            { "VitaminA", QuantityType.VitaminA },
-            { "Volume", QuantityType.Volume },
-            { "VolumeConcentration", QuantityType.VolumeConcentration },
-            { "VolumeFlow", QuantityType.VolumeFlow },
-            { "VolumeFlowPerArea", QuantityType.VolumeFlowPerArea },
-            { "VolumePerLength", QuantityType.VolumePerLength },
-            { "VolumetricHeatCapacity", QuantityType.VolumetricHeatCapacity },
-            { "WarpingMomentOfInertia", QuantityType.WarpingMomentOfInertia },
-        };
-
```
Comment (angularsen) [inline UnitsNet/GeneratedCode/Quantity.g.cs:272]: Good riddance!

**Review by angularsen — APPROVED** [review]
Looks great to me!

**Comment by angularsen** [comment]
Nuget should be out shortly
[Release UnitsNet/4.115.0 · angularsen/UnitsNet](https://github.com/angularsen/UnitsNet/releases/tag/UnitsNet%2F4.115.0)

**Pull request merged**